--- Glib-1.300/t/options.t.orig	2013-01-15 07:47:46.000000000 +0100
+++ Glib-1.300/t/options.t	2013-06-22 21:59:16.662022960 +0200
@@ -9,7 +9,7 @@
 unless (Glib -> CHECK_VERSION (2, 6, 0)) {
   plan skip_all => 'the option stuff is new in 2.6';
 } else {
-  plan tests => 33;
+  plan tests => 29;
 }
 
 # --------------------------------------------------------------------------- #
@@ -155,17 +155,6 @@
     is_deeply ($filename_array, [qw(/usr/bin/bla ./harness)]);
   }
 
-  # Test that there is no double-encoding for utf8-encoded strings.
-  {
-    @ARGV = qw(-s ❤ ❤);
-    $context -> parse();
-
-    is ($string, '❤');
-    is (length $string, 1);
-
-    is ($ARGV[0], '❤');
-    is (length $ARGV[0], 1);
-  }
 }
 
 # --------------------------------------------------------------------------- #
